packages/realm_dart/lib/src/init.dart

```diff
@@ -65,26 +58,39 @@ String get _exeDirName => p.dirname(Platform.resolvedExecutable);
 
 DynamicLibrary _openRealmLib() {
   final libName = _getLibName(realmBinaryName);
-  final root = _getNearestProjectRoot(Platform.script.path) ?? _getNearestProjectRoot(p.current);
-
-  // Try to open lib from various candidate paths
-  List<String> errMessages = [];
-  for (final open in [
-    () => _open(libName), // just ask OS..
-    () => _open(p.join(_exeDirName, libName)), // try finding it next to the executable
-    if (root != null) () => _open(p.join(root, 'binary', Platform.operatingSystem, libName)), // try finding it relative to project
-    () => _open(_getBinaryPath(libName)), // find it where it is installed by plugin
-  ]) {
+
+  DynamicLibrary? tryOpen(String path) {
```

Comment:
Would you say that there's a preference in Dart to have e.g. `tryOpen()` and `throwError()` local within `_openRealmLib()` rather than private functions defined outside of the parent function?

Reply:
No, I wouldn't per se.

It signals/conveys/ensures that these can (and will) only be called from within the declaring function, and it also allows the inner functions to refer to state in the outer.

I could use the same trick other places in this file, but I decided to try and keep the PR tractable instead.

Reply:
Alright 🙂